### ISSUE TITLE
Remember list type and time distribution precision for session

### ISSUE THREAD

**Issue by Mollerz** [cs0x7f/cstimer#74]
Just like each session remembers multi-phase, it would be good if it also remembered:
- List 1 type
- List 1 length
- List 2 type
- List 2 length
- time distribution precision

For example, for BLD and big cubes I might want to see mo3 and ao5, but I don't care about mo3 for a lot of events.

**Comment by topppits** [comment]
Thanks for the changes, they are awesome and I can't wait to use them :)

2 things I noticed in the /new version:
- typing in text fields in the options is no longer possible, which can be a bit annoying when I want to switch Voice volume from 100 to say 50 and I have to click 50 times. Left-click and holding down the mouse button also does only 1 step.
- for Statistical indicators I can use a different one per session, but only one custom one. I'd like to have mo3 ao5 ao12 ao100 ao1000 for pretty much all sessions but blind, where I'd like to have mo3 ao5 ao12. This is not possible. Both would need to be a custom option and when I change what "Custom" is for one session it changes for all sessions where I selected Custom.

**Comment by cs0x7f** [comment]
Thank you for your fast reply.

>     * typing in text fields in the options is no longer possible, which can be a bit annoying when I want to switch Voice volume from 100 to say 50 and I have to click 50 times. Left-click and holding down the mouse button also does only 1 step.
It is fixed in the latest commit. The bug is due to a previous update, which is for Ctrl+Alt+* keyboard shortcuts.


>     * for Statistical indicators I can use a different one per session, but only one custom one. I'd like to have mo3 ao5 ao12 ao100 ao1000 for pretty much all sessions but blind, where I'd like to have mo3 ao5 ao12. This is not possible. Both would need to be a custom option and when I change what "Custom" is for one session it changes for all sessions where I selected Custom.
> Also what's the "Function (toolsfunc):" under tools?
Well, some of options are internally stored without shown in the option dialog, such as scramble type, session data, custom statistics indicator values, selected functions in the tool panel, etc.
In current implementation, I make them visible but not editable in the option dialog, and then users are able to decide whether these values should be modified when switching sessions.

**Comment by topppits** [comment]
nice, text field entry works again.

Sorry but I don't fully understand the rest.

- could you make it possible for 
option > statistics > statistical indicators
to have different custom values for different sessions?

- option > tools > Selected Functions
this now has a checkbox on the right. What changes if I check it? Couldn't find anything.

- this is kind of OT in this thread, but can you switch the keyboard shortcuts:
alt + up/down -> next/last session
alt + up brings me to the next session, which in the list is below the current one. I think it'd be much more intuitive if alt + down would bring me to the session which is below in the session list and alt + up would get me to the session which is above in the list.

**Comment by cs0x7f** [comment]
I'm outside right now.

If an option is checked, its value will be stored in the session data. Then when you switch session, the option will be also changed.
To have different stat indicator, you may check both 'Custom Statistics Indicator' and 'Statistical Indicator'
To automatically switch tool functions for each session (e.g. cross solver for session 1, eoline solver for session 2), check 'Selected Functions'
It seems the new update is not as intuitive at all. I'll try to find an intuitive ways to tell user what does the checkbox means.

Shortcut:
Good suggestion. I will change it ASAP.

**Comment by topppits** [comment]
Thx! I understood the checkboxes, but probably only because I was aware that this new option could come. I just didn't understand what you meant by "Selected Functions", but I do now.

>To have different stat indicator, you may check both 'Custom Statistics Indicator' and 'Statistical Indicator'

It works that I can have different Statistical indicators for different sessions, but I can only choose between the 3 suggested options and one Custom option. What I want is a different Custom option for different sessions. I'll try to illustrate what I mean with an example.

session 1: 3x3
I want as Statistical Indicator mo3 ao5 ao12 ao50 ao100 ao1000, so I choose Custom and put that in.

session 2: 3BLD
I want as Statistical Indicator mo3 ao5 ao12, so I choose Custom and put that in.

Now when I go back to session 1: 3x3, it's still Custom for Statistical Indicator, but not what I entered before (mo3 ao5 ao12 ao50 ao100 ao1000), now it is also mo3 ao5 ao12. The Custom option can only be one thing.

**Comment by cs0x7f** [comment]
Well, yes, it's a bug that I have not found before.
The bug is due to that in previous versions, the internal stat indicator cannot be modified except selecting the 'Custom' item, so I add a cache in the statistics module. While right now it might be modified when session switching, but the cached value is not changed.

The bug is fixed in the latest commit. Also, the keyboard shortcut is swapped as your suggestion.

**Comment by topppits** [comment]
Weird little bug with the tool "Confirm time".

After checking the Select Functions checkbox under tools I can choose different tools for every session and this works fine until I choose "Confirm time" for a session. No when I switch sessions the tool will switch to Confirm time whichever tool I previously chose. Below the tool it'll still show the name of the tool I chose but the tool itself will always switch to Confirm time as soon as I switch sessions. When I go to the session where I chose the Confirm time tool and switch it to anything else it's working again as expected.
